btrfs/btrfs.go

```diff
@@ -517,3 +517,37 @@ func (b *btrfs) SetOverlayDirs(name string, overlayDirs types.OverlayDirs, layer
 	}
 	return errors.Errorf("Using overlay_dirs with btrfs storage is forbidden, use overlay storage instead")
 }
+
+func Check(config types.StackerConfig) error {
+	isBtrfs, err := DetectBtrfs(config.RootFSDir)
+	if err != nil {
+		return err
+	}
+
+	// it's already a mounted btrfs, nothing to worry about
+	if isBtrfs {
+		return nil
+	}
+
+	// can we mount btrfs filesystems? this is not allowed as an
+	// unprvileged user in any kernel, so we need to be euid 0 and btrfs
+	// needs to be listed in /proc/filesystems.
+	if os.Geteuid() != 0 {
+		return errors.Errorf("not root and %v is not btrfs", config.RootFSDir)
+	}
+
+	// btrfs might be built as a module, so let's modprobe; modprobe might
+	// fail if it was builtin, so let's just ignore modprobe's result
+	_ = exec.Command("modprobe", "btrfs").Run()
```

Comment:
just to avoid unnecessary failing modprobe, you could do this only if not already FilesystemIsSupported("btrfs").

```go

	if hasBtrfs, err = mount.FilesystemIsSupported("btrfs"); err != nil {
		return err
	} else if !hasBtrfs {
		_ = exec.Command("modprobe", "btrfs").Run()
		if hasBtrfs, err = mount.FilesystemIsSupported("btrfs"); err != nil {
			return err
		}
	}
```

Reply:
yep, that's a nice improvement, thanks.